src/credentials/GroupDataProviderImpl.cpp

```diff
@@ -830,11 +821,14 @@ struct KeySetData : PersistentData<kPersistentBufferMax>
                 ReturnErrorOnFailure(reader.Next(TagKeyHash()));
                 ReturnErrorOnFailure(reader.Get(key.hash));
                 // key value
-                ByteSpan value;
+                ByteSpan encryption_key;
                 ReturnErrorOnFailure(reader.Next(TagKeyValue()));
-                ReturnErrorOnFailure(reader.Get(value));
-                VerifyOrReturnError(Crypto::CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES == value.size(), CHIP_ERROR_INTERNAL);
-                memcpy(key.value, value.data(), value.size());
+                ReturnErrorOnFailure(reader.Get(encryption_key));
+                VerifyOrReturnError(Crypto::CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES == encryption_key.size(), CHIP_ERROR_INTERNAL);
+                memcpy(key.encryption_key, encryption_key.data(), encryption_key.size());
+                // Re-derive privacy key from encryption key when loading from storage to save on storage size.
+                MutableByteSpan privacy_key(key.privacy_key, Crypto::CHIP_CRYPTO_SYMMETRIC_KEY_LENGTH_BYTES);
```

Comment:
```suggestion
                MutableByteSpan privacy_key(key.privacy_key);
```
It's already got the right size, and this enforces that we don't get size mismatches between here and the declaration of `privacy_key`.